linode/manager
# refactor: [M3-8157] - Clean up images drawer

## Description 📝
Apply our latest coding standards to `ImagesDrawer` including:
* Separating `ImagesDrawer` into `EditImageDrawer` and `RestoreImageDrawer`
* Using `react-hook-form` instead of manually managing form state

## Changes  🔄
| File | Major Changes |
| --- | -----------------|
| [api-v4/src/images/images.ts](https://github.com/linode/manager/pull/10487/files#diff-86427837dd4bcd3095357f1a738b88195459f8341e85a82f1a5e3fbeba160567) | Simplify `updateImage` function signature. |
| [api-v4/src/images/types.ts](https://github.com/linode/manager/pull/10487/files#diff-cd50ed55a62f409baacf68865bb7467a4c8bb341d51b7df91ec2792f1be3bda2) | New `UpdateImagePayload` type. |
| [manager/src/features/Images/EditImageDrawer.test.tsx](https://github.com/linode/manager/pull/10487/files#diff-7d88720fe53d4409d062d2305c4b532e9276a54aaa8cca6a8f5d24c965c4d8c1) | Tests for `EditImageDrawer`. |
| [**manager/src/features/Images/EditImageDrawer.tsx**](https://github.com/linode/manager/pull/10487/files#diff-02c9ed9616afabcbbb8651100fdb5fe2ea1b20be106adb36b99770d91796f92f) | **Dedicated component for editing images, extracted from previous `ImagesDrawer`. Also implemented `react-hook-form` for state management.**|
| [manager/src/features/Images/ImageRow.tsx](https://github.com/linode/manager/pull/10487/files#diff-a73a8c6eca719347f26a9855036332bb5b5f1cfeee618d72da502d4c8c701420) | Simplified prop drilling. |
| [manager/src/features/Images/ImagesActionMenu.tsx](https://github.com/linode/manager/pull/10487/files#diff-bac967c3241541f709971f0ac48ff0b068a35819ec395b1770594b71a11a3ac6) | Updates to `Handlers` interface - they are now passed the entire `Image` |
| [**manager/src/features/Images/ImagesDrawer.tsx**](https://github.com/linode/manager/pull/10487/files#diff-b93f739b0452943d8e1d9b246ec3763b5115089907fda9b994ebf53038f94776)| **Replaced with `EditImageDrawer` and `RestoreImageDrawer`** |
| [**manager/src/features/Images/ImagesLanding.tsx**](https://github.com/linode/manager/pull/10487/files#diff-be57f661b259dbd940dfc604fcb8ea5272bfc07240e18445e31e1907767f2c47) | **Use new, separated drawer components. No longer stores drawer state.**|
| [manager/src/features/Images/RebuildImageDrawer.test.tsx](https://github.com/linode/manager/pull/10487/files#diff-897adb90c3088998f97c46c3484e52dd9b284b6e1d0999839bf1385de4ab8d03) | Tests for `RebuildImageDrawer`. |
| [**manager/src/features/Images/RebuildImageDrawer.tsx**](https://github.com/linode/manager/pull/10487/files#diff-3f903f5e6732d3fe77fb8c4c2d390f41f19ec6572b606f3e9306a9eff18b7276) | **Dedicated component for selecting images for rebuilding, extracted from previous `ImagesDrawer`. Also implemented `react-hook-form` for state management.**|
| [manager/src/features/Linodes/LinodesDetail/LinodeRebuild/RebuildFromImage.tsx](https://github.com/linode/manager/pull/10487/files#diff-bc83af921de116464abf8d2fd4e330ac3f88e69e07afb4fdde42dc47a6070f99) | Fix a bug where Image field was not being autopopulated from search params. |
| [manager/src/features/Linodes/LinodesDetail/LinodesDetail.tsx](https://github.com/linode/manager/pull/10487/files#diff-973cc4a54a9ef7bcef2e15ca2c27248c3d1e11c70022e572005f1540e1b7be7a) | Preserve search queries when redirecting. |
| [manager/src/queries/images.ts](https://github.com/linode/manager/pull/10487/files#diff-84ec4eeede5f9841b055dc0221689cc2453115c2273c116217cdebacf5991254) | New signature for `updateImage` function. |

## Preview 📷
*No visual changes.*

## How to test 🧪

* Verify Image Edit and Rebuild drawers work as expected:
   * Both drawers validate user inputs and display errors
   * Drawers reset when closing and re-opening
   * Submitting the Rebuild drawer redirects to the Linode Rebuild dialog with the Image select pre-populated

## As an Author I have considered 🤔

*Check all that apply*
